avocado/utils/ssh.py

```diff
@@ -248,3 +248,28 @@ def copy_files(self, source, destination, recursive=False):
             return result.exit_status == 0
         except process.CmdError as exc:
             raise NWException("failed to copy file {}".format(exc))
+
+    def check_connection(self):
+        """
+        Check for ssh connection if failed remove control path file
+        and reconnect it.
+
+        :returns: True if success or else False
+        :rtype: bool
+        """
+        if not self.connect():
+            if not self.quit():
+                for root, dirct, files in os.walk("/root/.ssh"):
+                    for file in files:
+                        if file.startswith("avocado-master-root"):
+                            path = os.path.join(root, file)
+                            os.remove(path)
+                if self.connect():
+                    return True
+                return False
+            else:
+                if self.connect():
+                    return True
+                return False
+        else:
+            return True
```

Comment:
This method is also returning a lot of true and false, with different situations. We could raise exceptions with explanations instead of true|false.